### PR TITLE
GalleryBlock: remove unused CSS and cleanup front-end.

### LINKED ISSUE #2272: Are the gallery css output on front margins needed
This is an issue purely with the front, rendered through a theme, not through the editor. Currently the gallery outputs a right margin on the front.:

`.wp-block-gallery:not(.components-placeholder) {
    margin-right: -16px;
}`

These work ok on fixed themes - with a single body width. However, I am working on for the Gutenberg Theme a fluid width, allowing for things like full width images/content to be up to the editor. This comes into more play as we move to have layout.

For example, this doesn't work because of that margin:

`.wp-block-gallery{
  margin: 1.5em auto;
  max-width: 1100px;
}
`

If you use a scary !important, it can be got around. I would suggest though that looking into if having a negative margin on the block is needed though. Happy if another work around also works. I mostly want to know why this is needed and can we avoid it?

### PR DESCRIPTION
**Pull request by mtias**
Closes #2272.

### CODE REVIEW

**Comment by jasmussen** [comment]
I don't know where the regression lies but I'm seeing some color issues with normal, hover and active states in the inserter:

screen shot 2017-08-08 at 18 17 44

screen shot 2017-08-08 at 18 17 52

Also seeing some weirdnesses with the image block, and inserting multiple galleries:

gallery

**Comment by mtias** [comment]
Maybe @aduth knows :)

**Comment by aduth** [comment]
Styling issues could indeed be related to #2160 because the "Portals" implementation there intentionally removes popover content from the CSS cascade, which is nice in that it avoids styles from a parent from affecting the popover content, but not so great for "global" styles.

One option is to render the portal into the base of the Gutenberg editor rather than the `body`.

File: blocks/library/gallery/index.js [hunk 1]
```diff
@@ -168,6 +168,7 @@ registerBlockType( 'core/gallery', {
 
 	save( { attributes } ) {
 		const { images, columns = defaultColumnsNumber( attributes ), align = 'none', imageCrop = true, linkTo = 'none' } = attributes;
+
```
Comment (aduth) [inline blocks/library/gallery/index.js]: This is probably going to cause a conflict for me in #1905 😬

File: blocks/library/gallery/style.scss [hunk 2]
```diff
@@ -1,23 +1,6 @@
...
-
-	.dashicon {
-		margin-right: 1ch;
-	}
-}
-
-.blocks-gallery__placeholder-instructions {
-	margin: 1.8em 0;
+.wp-block-gallery:not( .components-placeholder ) {
+	margin-right: -16px;
```
Comment (aduth) [inline blocks/library/gallery/style.scss]: Sorta feel the excess margin on gallery images could be better handled by an `:nth-of-type( 2n ) { margin-right: 0; }` etc.

**Review by aduth — APPROVED** [review]

**Pull request merged**